enterprise/server/atime_updater/atime_updater.go

```diff
@@ -0,0 +1,236 @@
+}
+
+func (u *atimeUpdater) Enqueue(ctx context.Context, instanceName string, digests []*repb.Digest, digestFunction repb.DigestFunction_Value) {
+	if len(digests) == 0 {
+		return
+	}
+
+	groupID := u.groupID(ctx)
+	jwt := u.authenticator.TrustedJWTFromAuthContext(ctx)
+	u.mu.Lock()
+	updates, ok := u.updates[groupID]
+	if !ok {
+		updates = &atimeUpdates{}
+		u.updates[groupID] = updates
+	}
+	u.mu.Unlock()
+
+	// Uniqueify the incoming digests. Note this wrecks the order of the
+	// requested updates. Too bad.
+	keys := map[digest.Key]struct{}{}
+	for _, digestProto := range digests {
+		keys[digest.NewKey(digestProto)] = struct{}{}
+	}
+
+	// TODO(iain): this locking may be a bit overzealous. Hopefully not though.
+	updates.mu.Lock()
+	defer updates.mu.Unlock()
+
+	// Always use the most recent JWT for a group for remote atime updates.
+	updates.jwt = jwt
+
+	// First, find the update that the new digests can be merged into, or create
+	// a new one if one doesn't exist.
+	var pendingUpdate *atimeUpdate
+	for _, update := range updates.updates {
+		if update.instanceName == instanceName && update.digestFunction == digestFunction {
+			pendingUpdate = update
+			break
+		}
+	}
+	if pendingUpdate == nil {
+		if len(updates.updates) >= u.maxUpdatesPerGroup {
+			log.CtxInfof(ctx, "Too many pending FindMissingBlobsRequests for updating remote atime for group %s, dropping some pending atime updates", groupID)
+			return
+		}
+		pendingUpdate = &atimeUpdate{
+			instanceName:   instanceName,
+			digests:        map[digest.Key]struct{}{},
+			digestFunction: digestFunction,
+		}
+		updates.updates = append(updates.updates, pendingUpdate)
+	}
+
+	// Now merge the new digests into the update.
+	for key := range keys {
+		if len(pendingUpdate.digests) >= u.maxDigestsPerUpdate {
+			log.CtxInfof(ctx, "FindMissingBlobsRequest for updating remote atime for group %s too large, dropping some pending atime updates", groupID)
+			break
+		}
+		pendingUpdate.digests[key] = struct{}{}
+	}
+}
+
+func (u *atimeUpdater) EnqueueByResourceName(ctx context.Context, downloadString string) {
+	if digest.IsActionCacheResourceName(downloadString) {
+		// ActionCache entries are always read authoritatively, so no need to
+		// enqueue an atime update.
+		return
+	}
+	rn, err := digest.ParseDownloadResourceName(downloadString)
+	if err != nil {
+		log.Warningf("Skipping remote atime update for malformed download resource name: %s [%s]", downloadString, err)
+		return
+	}
+	u.Enqueue(ctx, rn.GetInstanceName(), []*repb.Digest{rn.GetDigest()}, rn.GetDigestFunction())
+}
+
+func (u *atimeUpdater) EnqueueByFindMissingRequest(ctx context.Context, req *repb.FindMissingBlobsRequest) {
+	u.Enqueue(ctx, req.InstanceName, req.BlobDigests, req.DigestFunction)
+}
+
+func (u *atimeUpdater) start(opts Opts) {
+	tickerChan := opts.TickerForTesting
+	if tickerChan == nil {
+		ticker := time.NewTicker(u.updateInterval)
+		tickerChan = ticker.C
+		defer ticker.Stop()
+	}
```

Comment:
would it work to add a `TickerForTestingOnly() chan time.Time` method to atimeUpdate and grab the one you create here? Then you could omit Opts